synchronization/trigger.go

```diff
@@ -1,19 +1,20 @@
 package synchronization
 
 import (
+	"context"
 	"sync"
 	"sync/atomic"
 	"time"
 )
 
 type Trigger interface {
-	Start()
-	Reset(duration time.Duration)
+	Start(ctx context.Context)
+	Reset(ctx context.Context, duration time.Duration)
 	TimesTriggered() uint64
 	TimesReset() uint64
 	TimesTriggeredManually() uint64
 	IsRunning() bool
-	FireNow()
+	FireNow(ctx context.Context)
```

Comment:
same error prone issue as above; furthermore - why do we need `FireNow` at all?

Reply:
context issue is addressed above, but yes, right now i do not think anyone calls FireNow(), not really sure why we still need it other then 'its there and tested...'
the story ofc is 'i need the action to happen right now and not periodically, and i would like to begin waiting for the periodic time again'